### DISCUSSION TITLE
Support nested routing with saved state using a BottomNavigationBar

### DISCUSSION THREAD

**Discussion by sunndog** [csells/go_router#78] [General]
Using Material Design's Bottom Navigation [behaviour description](https://material.io/components/bottom-navigation#behavior) I would like to use go_router for "an Android app that requires frequent switching between sections [that] can preserve each section’s state".  Would like to be able to use go_router for deep linking and navigation between sections and the creation of individual page stacks where required.

The CupertinoTabView implementation currently uses a Navigator v1 component internally for each bottom navigation item and probably isn't a candidate for use with go_router.

**Comment by csells** [comment]
I think the idea would be to have a separate stack for each section? I'm having trouble even wrapping my head around that...

**Reply by sunndog** [reply]
The behaviour is ubiquitous on iOS but definitely becoming more common on Android.  In our case we are migrating from a native Android and native iOS application to Flutter and our ux designers are looking for this behaviour across platforms.  As a result we're levarging the Cupertino components for the bottom navigation but generally sticking to a branded Material component infrastructure for everything else.  Our navigation right now is a little too imperative for my liking and a solution like yours could help solve our problem.

**Reply by csells** [reply]
Do you have a minimal sample implantation of the kind of thing you're talking about? I'd be interested in taking it apart and understanding whether it's something go_router could do.

**Reply by davidmartos96** [reply]
@csells Interested in this feature as well. Our Flutter app uses this kind of navigation with the navigator 1.0.
It can easily be done with the help of Offstage and Stack. Here is a medium from Andreas explaining the approach.
https://medium.com/coding-with-flutter/flutter-case-study-multiple-navigators-with-bottomnavigationbar-90eb6caa6dbf

There is also another great article from Hans Muller about the topic.

**Reply by csells** [reply]
that is an excellent example! I've logged it with a feature request here: https://github.com/csells/go_router/issues/82

**Comment by csells** [comment]
I think it would be useful to add such functionality to go_router for sure. I'm curious if you try it where go_router falls down and how you think it could be improved.

**Comment by percula** [comment]
There's been a bunch of updates to go_router since this discussion, but I don't see an example that shows using BottomNavigationBar. Is this possible?

**Reply by csells** [reply]
there aren't any specific examples that show the use of `BottomNavigatorBar`. Have you tried updating [the nested navigation sample](https://github.com/csells/go_router/blob/main/go_router/example/lib/nested_nav.dart) to use `BottomNavigatorBar` instead of `TabBar`?

**Comment by mark8044** [comment]
One thing to keep in mind, when using stacked navigation in a bottom navigation bar, as things stand, its very difficult-to-impossible to handle Primary Scroll Controller taps on the status bar.  So for iOS apps you lose the coveted tap status bar scroll list view to the top feature when using many stacked navigation arrangements.  

I'm hoping to start using go_router this weekend and see if it can help in this regard, but incase it doesn't, this is an important thing to consider.

**Comment by tananga** [comment]
@csells  I think that people want to know how to implement something like this https://youtu.be/9oH42_Axr3Q

**Reply by mark8044** [reply]
Yes exactly right
